examples/bot_error_handling.py

```diff
@@ -0,0 +1,92 @@
+    # this event is called every time an exception occurs during the "divide" command's processing.
+    # may this be cause by parsing errors (e.g. invalid quotes or a converter raising the exception),
+    # the command being on cooldown, disabled or a general invoke error.
+    # Note: the global command error handler (see below) will be called after.
+    #
+    # ctx   = the command's context
+    # error = the exception raised
+
+    # all exceptions that do not derive from CommandError are wrapped
+    # into CommandInvokeError and stored in its 'original' attribute,
+    # so we unwrap it.
+    if isinstance(error, commands.CommandInvokeError):
+        error = getattr(error, "original")
+
+    # respond with an error message if the user tries do divide by zero
+    if isinstance(error, ZeroDivisionError):
+        await ctx.send("You can't divide a number by zero.")
+
+@bot.event
+async def on_command_error(ctx, error):
+    # this event is called every time an exception occurs during a command's processing.
+    # may this be caused by parsing errors (e.g. invalid quotes or a converter raising the exception),
+    # a command being on cooldown, disabled, not found or a general invoke error.
+    #
+    # ctx   = the command's context
+    # error = the exception raised
+
+    # this checks if the command has a local error handler (see above),
+    # if so skip all the below handling.
+    if hasattr(ctx.command, "on_error"):
+        return
+
+    if isinstance(error, commands.CommandInvokeError):
+        error = getattr(error, "original")
+
+    # do nothing since we don't really care in this case
+    if isinstance(error, commands.CommandNotFound):
+        return
+
+    # as the name says, this catches all errors derived from user input.
+    elif isinstance(error, commands.UserInputError):
+        # this is fine for most use cases since it replies with a friendly error message.
+        await ctx.send(error)
+
+    # the command is on cooldown
+    elif isinstance(error, commands.CommandOnCooldown):
+        # we can easily access the exception and context's attributes to give a more helpful error message.
+        msg = "The \"{0}\" command is on cooldown, wait {1:.2f} seconds.".format(ctx.command.name, error.retry_after)
+        await ctx.send(msg)
+
+    # here you can handle all other types of exceptions.
+
+    else:
+        await ctx.send("An unexpected error has occurred and my developer has been notified, sorry.")
+        # make sure to always print the exception's traceback at the end of the event, else it will be "eaten".
+
+        # Note: it's impossible to use traceback.print_exc() or traceback.format_exc() due to
+        # how event dispatching messes with the exception context.
+
+        traceback.print_exception(type(error), error, error.__traceback__)
+
+# Treat your bot's token like your account's password,
+# never share it with anyone or upload it anywhere.
+bot.run("TOKEN")
```

Comment:
Tracebacks with logging in generally show this line. It may be preferred to have it in another line instead, eg 
```python
token = "TOKEN"
bot.run(token)
```

Reply:
Everything should be fixed up, thanks.